File: elecciones_app_movil/lib/data/model/candidato/acta_dignidad.dart
import 'package:elecciones_app_movil/data/model/candidato/dignidad_ubicacion.dart';

class ActaDignidad {
  ActaDignidad(
      {required this.id,
      required this.dignidadUbicacion,
      required this.identificacion,
      required this.actaArchivo,
      required this.actaNombre,
      required this.actaDescripcion,
      required this.escrutada});

  int id;
  DignidadUbicacion dignidadUbicacion;
  String identificacion;
  dynamic actaArchivo;
  String actaNombre;
  String actaDescripcion;
  int escrutada;

  ActaDignidad copyWith(
          {int? id,
          DignidadUbicacion? dignidadUbicacion,
          String? identificacion,
          dynamic actaArchivo,
          String? actaNombre,
          String? actaDescripcion,
          int? escrutada}) =>
      ActaDignidad(
          id: id ?? this.id,
          dignidadUbicacion: dignidadUbicacion ?? this.dignidadUbicacion,
          identificacion: identificacion ?? this.identificacion,
          actaArchivo: actaArchivo ?? this.actaArchivo,
          actaNombre: actaNombre ?? this.actaNombre,
          actaDescripcion: actaDescripcion ?? this.actaDescripcion,
          escrutada: escrutada ?? this.escrutada);

  factory ActaDignidad.fromMap(Map<String, dynamic> json) => ActaDignidad(
      id: json["id"],
      dignidadUbicacion: DignidadUbicacion?.fromMap(json["dignidadUbicacion"]),
      identificacion: json["identificacion"],
      actaArchivo: json["actaArchivo"],
      actaNombre: json["actaNombre"],
      actaDescripcion: json["actaDescripcion"],
      escrutada: json["escrutada"]);

  Map<String, dynamic> toMap() => {
        "id": id,
        "dignidadUbicacion": dignidadUbicacion?.toMap(),
        "identificacion": identificacion,
        "actaArchivo": actaArchivo,
        "actaNombre": actaNombre,
        "actaDescripcion": actaDescripcion,
        "escrutada": escrutada
      };

  @override
  String toString() {
    return 'ActaDignidad{id: $id, dignidadUbicacion: $dignidadUbicacion, identificacion: $identificacion, actaArchivo: $actaArchivo, actaNombre: $actaNombre, actaDescripcion: $actaDescripcion, escrutada: $escrutada}';
  }
}

File: elecciones-api-cliente/src/main/java/com/eleccciones/cliente/data/entities/candidatos/ActaDignidad.java
package com.eleccciones.cliente.data.entities.candidatos;

import com.eleccciones.cliente.data.entities.core.impl.AbstractEntity;
import com.eleccciones.cliente.data.entities.ubicacion.Junta;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.concurrent.locks.StampedLock;

@Entity
@Table(name = "acta_dignidad")
@Getter
@Setter
public class ActaDignidad extends AbstractEntity {

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    @JoinColumn(name = "id_junta", referencedColumnName = "id")
    private Junta junta;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "id_dignidad_ubicacion", referencedColumnName = "id")
    private DignidadUbicacion dignidadUbicacion;

    @Column(name = "identificacion")
    private String identificacion;

    @Lob
    @Type(type = "org.hibernate.type.ImageType")
    @Column(name = "acta_archivo")
    private byte[] actaArchivo;

    @Column(name = "acta_nombre")
    private String actaNombre;

    @Column(name = "descripcion")
    private String actaDescripcion;

    @Column(name = "escrutada")
    private Integer escrutada;

    @Transient
    @JsonIgnore
    private StampedLock lock;

    public ActaDignidad() {
        this.lock = new StampedLock();
    }
}
